### ISSUE TITLE
Folder > frontend_dist - Not Found (When Upgrading from 1.17.0 to 2.0.0-beta.3)

### ISSUE THREAD

**Issue by michael-v-bloom** [modoboa/modoboa#2465]
# Impacted versions

* OS Type: Ubuntu
* OS Version: 20.04 (LTS) x64
* Database Type: MySQL
* Database version: 15.1
* Modoboa: 1.17.0
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce
We're trying to upgrade our current version of Modoboa to the Beta version with new Admin interface.

So, we followed the steps from the upgrade guide:

First we activate the dev environment:
`> source /srv/modoboa/env/bin/activate`

Then we execute the commands for the upgrade:
`> pip install modoboa==2.0.0-beta.3`
`> cd /srv/modoboa/instance`

Then we follow the next step:
`> python manage.py migrate`   

The output is:
```
Operations to perform:
  Apply all migrations: admin, auth, authtoken, contenttypes, core, dnstools, lib, limits, maillog, modoboa_amavis, modoboa_contacts, modoboa_postfix_autoreply, modoboa_radicale, otp_static, otp_totp, relaydomains, reversion, sessions, sites, transport
Running migrations:
  No migrations to apply. 
```

Then we do:
`> python manage.py collectstatic`

The output is:
```
You have requested to collect static files at the destination
location as specified in your settings:

    /srv/modoboa/instance/sitestatic

This will overwrite existing files!
Are you sure you want to do this?

Type 'yes' to continue, or 'no' to cancel:
```

We type YES and the output is:
```
Type 'yes' to continue, or 'no' to cancel: yes
Found another file with the destination path 'index.html'. It will be ignored since only the first encountered file is collected. If this is not what you want, make sure every static file has a unique path.

0 static files copied to '/srv/modoboa/instance/sitestatic', 3150 unmodified.
(env) dmail@mail:/srv/modoboa/instance$ python manage.py check --deploy
System check identified some issues:

WARNINGS:
?: (modoboa-amavis.W001) AMAVIS_DEFAULT_DATABASE_ENCODING does not match the character encoding used by the Amavis database.
	HINT: Check your database character encoding and set/update AMAVIS_DEFAULT_DATABASE_ENCODING.
?: (security.W001) You do not have 'django.middleware.security.SecurityMiddleware' in your MIDDLEWARE so the SECURE_HSTS_SECONDS, SECURE_CONTENT_TYPE_NOSNIFF, SECURE_BROWSER_XSS_FILTER, and SECURE_SSL_REDIRECT settings will have no effect.

System check identified 2 issues (3 silenced).
```

Then we go to the version specific instructions where we need to copy Frontend files from **frontend_dist** to new **frontend** folder:
`> mkdir /srv/modoboa/instance/frontend`
`> cp -r /srv/modoboa/env/lib/python3.8/site-packages/modoboa/frontend_dist/* /srv/modoboa/instance/frontend`

But, the output is that the folder **/srv/modoboa/env/lib/python3.8/site-packages/modoboa/frontend_dist/** doesn't exists... 

# Current behavior

**/srv/modoboa/env/lib/python3.8/site-packages/modoboa/frontend_dist/** doesn't exists

Did we missed something? We checked all other issues and discussions related to this and other upgrades, but we've found nothing that could help. Any ideas @tonioo what it could be?

Thank you.
Michael

**Comment by tonioo** [comment]
@michael-v-bloom Can you list the folder content please?

**Comment by michael-v-bloom** [comment]
@tonioo 
Here are the contents for:  **/srv/modoboa/env/lib/python3.8/site-packages/modoboa**
```
__init__.py                          
__pycache__                 
admin                            
bower_components
core  lib       
dnstools 
parameters  
ldapsync    
limits
locale
maillog
policyd
relaydomains
static
test_settings.py
transport
templates     
urls.py
urls_api.py
```

**Comment by nike-on** [comment]
I also wanted to update from 1.17.0 to the 2.0.0 Beta 3 version on clean install of Debian 10.
and as long as the download process ran smoothly
```bash
(env) $ pip install modoboa==2.0.0-beta.3
[....]
Successfully installed PyYAML-6.0 aioredis-1.3.1 attrs-21.4.0 django-ckeditor-6.1.0 
django-otp-1.0.6 django-phonenumber-field-5.2.0 django-reversion-3.0.9 
djangorestframework-simplejwt-4.8.0 dnspython-2.1.0 drf-flex-fields-0.9.8 
drf-spectacular-0.22.0 feedparser-6.0.8 gevent-21.1.2 greenlet-1.1.0 hiredis-2.0.0 
importlib-resources-5.7.1 inflection-0.5.1 jsonfield-3.1.0 jsonschema-4.4.0 
modoboa-2.0.0b3 pyjwt-2.3.0 pyrsistent-0.18.1 sgmllib3k-1.0.0
```

So the migration process already came out with error
```bash 
(env) $ python manage.py migrate
Warning #0: @extend_schema_view argument "retrieve" was not found on view DomainViewSet. method override for "retrieve" will be ignored.
Warning #1: @extend_schema_view argument "create" was not found on view DomainViewSet. method override for "create" will be ignored.
Operations to perform:
  Apply all migrations: admin, auth, authtoken, contenttypes, core, dnstools, lib, limits, maillog, modoboa_amavis, modoboa_contacts, modoboa_postfix_autoreply, modoboa_radicale, otp_static, otp_totp, relaydomains, reversion, sessions, sites, transport
Running migrations:
  Applying admin.0018_auto_20201204_0935... OK
  Applying admin.0019_update_disabled_accounts_aliases... OK
  Applying admin.0020_auto_20210130_2226...Traceback (most recent call last):
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.FeatureNotSupported: cannot alter type of a column used by a view or rule
DETAIL:  rule _RETURN on view dkim depends on column "name"


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    main()
  File "manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 234, in handle
    fake_initial=fake_initial,
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 249, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 535, in alter_field
    old_db_params, new_db_params, strict)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/postgresql/schema.py", line 124, in _alter_field
    new_db_params, strict,
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 685, in _alter_field
    params,
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 137, in execute
    cursor.execute(sql, params)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/srv/modoboa/env/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.NotSupportedError: cannot alter type of a column used by a view or rule
DETAIL:  rule _RETURN on view dkim depends on column "name"

(env) $
```

**Comment by tonioo** [comment]
@michael-v-bloom Sorry for the late reply. I just tried to install the beta3 package in a test virtualenv and I can see the frontend_dist folder as expected...
I'm using python 3.9 with an up-to-date pip. Have you tried to update pip?

**Comment by tonioo** [comment]
And I used this command:
```
pip install --pre modoboa
```

**Comment by cremesk** [comment]
hi,

i upgrade my modoboa today to version 2.1.0 .
it looks good.. but i'm not able to upgrade the frontend..

https://modoboa.readthedocs.io/en/latest/upgrade.html#new-admin-interface

`frontend_dist/` doesn't exists in `/srv/modoboa/env/lib/python3.9/site-packages/modoboa`

```
ls -lah  /srv/modoboa/env/lib/python3.9/site-packages/modoboa
total 112K
drwxr-xr-x  21 modoboa modoboa 4.0K Apr 22 08:40 .
drwxr-xr-x 191 modoboa modoboa  12K Apr 22 08:40 ..
-rw-r--r--   1 modoboa modoboa  236 Apr 22 08:40 __init__.py
drwxr-xr-x   2 modoboa modoboa 4.0K Apr 22 08:40 __pycache__
drwxr-xr-x  14 modoboa modoboa 4.0K Apr 22 08:40 admin
drwxr-xr-x  14 modoboa modoboa 4.0K Apr 22 08:40 bower_components
drwxr-xr-x  15 modoboa modoboa 4.0K Apr 22 08:40 core
drwxr-xr-x   9 modoboa modoboa 4.0K Apr 22 08:40 dmarc
drwxr-xr-x   7 modoboa modoboa 4.0K Apr 22 08:40 dnstools
drwxr-xr-x   7 modoboa modoboa 4.0K Apr 22 08:40 imap_migration
drwxr-xr-x   4 modoboa modoboa 4.0K Apr 22 08:40 ldapsync
drwxr-xr-x   6 modoboa modoboa 4.0K Apr 22 08:40 lib
drwxr-xr-x   8 modoboa modoboa 4.0K Apr 22 08:40 limits
drwxr-xr-x  31 modoboa modoboa 4.0K Apr 22 08:40 locale
drwxr-xr-x  10 modoboa modoboa 4.0K Apr 22 08:40 maillog
drwxr-xr-x   5 modoboa modoboa 4.0K Apr 22 08:40 parameters
drwxr-xr-x   5 modoboa modoboa 4.0K Apr 22 08:40 pdfcredentials
drwxr-xr-x   5 modoboa modoboa 4.0K Apr 22 08:40 policyd
drwxr-xr-x   6 modoboa modoboa 4.0K Apr 22 08:40 relaydomains
drwxr-xr-x   5 modoboa modoboa 4.0K Apr 22 08:40 static
drwxr-xr-x   4 modoboa modoboa 4.0K Apr 22 08:40 templates
-rw-r--r--   1 modoboa modoboa 2.1K Apr 22 08:40 test_settings.py
drwxr-xr-x   8 modoboa modoboa 4.0K Apr 22 08:40 transport
-rw-r--r--   1 modoboa modoboa 3.6K Apr 22 08:40 urls.py
-rw-r--r--   1 modoboa modoboa  474 Apr 22 08:40 urls_api_v1.py
-rw-r--r--   1 modoboa modoboa  809 Apr 22 08:40 urls_api_v2.py
```

**Comment by Spitfireap** [comment]
Hi, we are having an issue with the deployment, a patch is being worked on.

**Comment by cremesk** [comment]
fixed in 2.1.1. thx!
